### PR TITLE
Add REST endpoints for filtering samples in metadata by age and BMI

### PR DESCRIPTION
**Pull request by gwarmstrong**
This PR includes a draft of the swagger yaml and tests for responses that should allow for API calls like:
```
GET …/metadata/category/values/age_cat
```
which would return something like “20s, 30s, 40s…”, the specific values of the category
and
```
GET …/metadata/sample-ids/age_cat=<a-specific-value>&bmi_cat=<a-specific-value>
```
which would return the sample ids that meet the age and bmi criteria

### CODE REVIEW

**Comment by gwarmstrong** [comment]
After taking a looks at the data it looks like I should rework this PR so that it uses `bmi_cat` instead of `bmi`

**Comment by gwarmstrong** [comment]
Alright that should be properly refactored now

File: microsetta_public_api/api/tests/test_api.py [hunk 1]
```diff
@@ -4,6 +4,237 @@
...
+                'sample_ids': [
+                    'sample-1',
+                ],
+                'some-extra-cat': ['stuff'],
+            })
+
+        _, self.client = self.build_app_test_client()
+        response = self.client.get(
+            "/api/metadata/sample-ids?age_cat=30s&bmi_cat=normal")
+        self.assertStatusCode(500, response)
```
Comment (wasade) [inline microsetta_public_api/api/tests/test_api.py:150]: We really shouldn't ever be throwing `500`s... is this something expected that can happen?

Comment (gwarmstrong) [reply microsetta_public_api/api/tests/test_api.py:150]: My understanding is that if your operation gives a response that does not follow the schema for your API, the client will give a 500 error. So these types of checks for 500's are generally verifying that certain types of outputs are disallowed by my API spec.

**Pull request merged**